core/src/inputs/mod.rs

```diff
@@ -49,7 +52,7 @@ impl ProgramInputs {
     /// - The number initial stack values is greater than 16.
     /// - Any of the initial stack values or the advice tape values are not valid field elements.
     /// - Any of the advice sets have the same root.
-    pub fn new(
+    pub fn new_with_vec(
```

Comment:
I would not change the name of the constructor. This will minimize changes elsewhere.